modules/sharedaddy/recaptcha.php

```diff
@@ -0,0 +1,168 @@
+		'invalid-input-response' => 'The response parameter is invalid or malformed',
+	);
+
+	/**
+	 * Create a configured instance to use the reCAPTCHA service.
+	 *
+	 * @param string $site_key   Site key to use in HTML code.
+	 * @param string $secret_key Shared secret between site and reCAPTCHA server.
+	 * @param array  $config     Config array to optionally configure reCAPTCHA instance.
+	 */
+	public function __construct( $site_key, $secret_key, $config = array() ) {
+		$this->site_key   = $site_key;
+		$this->secret_key = $secret_key;
+		$this->config     = wp_parse_args( $config, $this->get_default_config() );
+	}
+
+	/**
+	 * Get default config for this reCAPTCHA instance.
+	 *
+	 * @return array Default config
+	 */
+	public function get_default_config() {
+		return array(
+			'language'       => get_locale(),
+			'tag_class'      => 'g-recaptcha',
+			'tag_attributes' => array(
+				'theme'    => 'light',
+				'type'     => 'image',
+				'tabindex' => 0,
+			)
+		);
+	}
+
+	/**
+	 * Calls the reCAPTCHA siteverify API to verify whether the user passes
+	 * CAPTCHA test.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-respones' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return bool|WP_Error Returns true if verified. Otherwise WP_Error is returned.
+	 */
+	public function verify( $response, $remote_ip ) {
+		// No need make a request if response is empty.
+		if ( empty( $response ) ) {
+			return new WP_Error( 'missing-input-response', $this->error_codes['missing-input-response'], 400 );
+		}
+
+		$resp = wp_remote_post( self::VERIFY_URL, $this->get_verify_request_params( $response, $remote_ip ) );
+		if ( is_wp_error( $resp ) ) {
+			return $resp;
+		}
+
+		$resp_decoded = json_decode( wp_remote_retrieve_body( $resp ), true );
+		if ( ! $resp_decoded ) {
+			return new WP_Error( 'invalid-json', 'Invalid JSON', 400 );
+		}
+
+		if ( isset( $resp_decoded['success'] ) && true === $resp_decoded['success'] ) {
+			return true;
+		}
+
+		$error_message = 'Unexpected response';
+		$error_code    = 'unexpected-response';
+
+		// Use the first error code if exists.
+		if ( isset( $resp_decoded['error-codes'] ) && is_array( $resp_decoded['error-codes'] ) ) {
+			if ( isset( $resp_decoded['error-codes'][0] ) && isset( $this->error_codes[ $resp_decoded['error-codes'][0] ] ) ) {
+				$error_message = $this->error_codes[ $resp_decoded['error-codes'][0] ];
+				$error_code    = $resp_decoded['error-codes'][0];
+			}
+		}
+
+		return new WP_Error( $error_code, $error_message, 400 );
+	}
+
+	/**
+	 * Get siteverify request parameters.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-respones' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return array
+	 */
+	public function get_verify_request_params( $response, $remote_ip ) {
```

Comment:
I don't see urgency to make it private. The reason it's public is I wanted it to have unit test.